aten/src/ATen/core/StorageImpl.h

```diff
@@ -74,6 +74,8 @@ struct CAFFE2_API StorageImpl : public c10::intrusive_ptr_target {
 
   template <typename T>
   inline T* data() const {
+    // TODO: This is bad: it means storage.data<T>() calls only work on
```

Comment:
Yes, this is something we should fix in short order.

Reply:
Like, we'll have a useful merge even if this doesn't work, but it will be even better if all C++ side APIs handle non-fundamental types correctly. But this will require work, and some coding (which may better be done after some codepaths get deduplicated.)

Reply:
Yes, should be.